src/main/java/com/glisco/deathlog/storage/BaseDeathLogStorage.java

```diff
@@ -57,6 +71,11 @@ protected void load(File file, List<DeathInfo> into) {
     }
 
     protected void save(File file, List<DeathInfo> list) {
+        // Copying the list because of concurrency issues
+        FS_EXECUTOR.execute(() -> save0(file, ImmutableList.copyOf(list)));
```

Comment:
Why are you copying it in the lambda and not in the outer method?

Reply:
Smaller diff + you need less indents in the save0 method.

Reply:
Well yes, but then you're going to copy the list in the thread pool, which can cause concurrency issues

Reply:
Oh, sorry, I misunderstood your question, will fix it.